semanticdb/semanticdb3/javacp.md

```diff
@@ -0,0 +1,480 @@
+    <td>Explained below on per-definition basis.</td>
+  </tr>
+  <tr>
+    <td><code>accessibility</code></td>
+    <td>Explained below on per-definition basis.</td>
+  </tr>
+  <tr>
+    <td><code>owner</code></td>
+    <td>See <a href="#java-symbol">Symbol</a>.</td>
+  </tr>
+</table>
+
+**Class declarations** [\[76\]][76] are represented by a single symbol with the `CLASS` kind.
+
+```java
+package a;
+class C extends S1 implements I {
+  String m1;
+  static unit m2();
+  int m3(String e) throws E;
+  static class D1<T1 extends S2 & S3, T2> { }
+  class D2 { }
+}
+```
+
+<table>
+  <tr>
+    <td><b>Definition</b></td>
+    <td width="275px"><b>Symbol</b></td>
+    <td><b>Kind</b></td>
+    <td><b>Signature</b></td>
+  </tr>
+  <tr>
+    <td><code>C</code></td>
+    <td><code>a.C#</code></td>
+    <td><code>CLASS</code></td>
+    <td><code>ClassInfoType(List(), List(&lt;S1&gt;, &lt;I&gt;), List(&lt;a.C#m1&gt;, &lt;a.C#m3(String).&gt;, &lt;a.C#D2#&gt;, &lt;a.C#m2().&gt;, &lt;a.C#D1#&gt;))</code></td>
+  </tr>
+  <tr>
+    <td><code>m1</code></td>
+    <td><code>a.C#m1.</code></td>
+    <td><code>FIELD</code></td>
+    <td><code>TypeRef(None, &lt;java.lang.String&gt;, List())</code></td>
+  </tr>
+  <tr>
+    <td><code>m2</code></td>
+    <td><code>a.C#m2().</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List() TypeRef(None, &lt;scala.Unit#&gt;))</code></td>
+  </tr>
+  <tr>
+    <td><code>m3</code></td>
+    <td><code>a.C#m3(String).</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(&lt;a.C#m3(String).(e)&gt;) TypeRef(None, &lt;scala.Int#&gt;))</code></td>
+  </tr>
+  <tr>
+    <td><code>e</code></td>
+    <td><code>a.C#m3(String).(e)</code></td>
+    <td><code>PARAMETER</code></td>
+    <td><code>TypeRef(None, &lt;java.lang.String#&gt;, List())</code></td>
+  </tr>
+  <tr>
+    <td><code>e</code></td>
+    <td><code>a.C#D1#[T1]</code></td>
+    <td><code>TYPE_PARAMETER</code></td>
+    <td><code>TypeType(List(), None, Some(IntersectionType(List(&lt;S2&gt;, &lt;S2&gt;))))</code></td>
+  </tr>
+  <tr>
+    <td><code>e</code></td>
+    <td><code>a.C#D1#[T2]</code></td>
+    <td><code>TYPE_PARAMETER</code></td>
+    <td><code>TypeType(List(), None, None)</code></td>
+  </tr>
+  <tr>
+    <td><code>D1</code></td>
+    <td><code>a.C#D1#</code></td>
+    <td><code>CLASS</code></td>
+    <td><code>ClassInfoType(List(&lt;a.C#D1#[T1]&gt;, &lt;a.C#D1#[T2]&gt;), List(&lt;java.lang.Object#&gt;), List())</code></td>
+  </tr>
+  <tr>
+    <td><code>D2</code></td>
+    <td><code>a.C#D2#</code></td>
+    <td><code>CLASS</code></td>
+    <td><code>ClassInfoType(List(), List(), List())</code></td>
+  </tr>
+</table>
+
+Notes:
+* Class members must appear in the following order:
+  * non-static members first, following the same order as they appear in the original source
+  * static members secondly, following the same order as they appear in the original source
```

Comment:
Why was that necessary? Not sure I can recall.

Reply:
Right, now I remember.

Reply:
How about we move the requirement about order to the definition of disambiguator? Order of symbols in class info types doesn't need to be synchronized between metac and javacp - only the order of overloads does.